.github/workflows/ci.yml

```
@@ -8,35 +8,48 @@ on:
 jobs:
   ci_suite:
     runs-on: ubuntu-latest
-    env:
-      POETRY_VIRTUALENVS_CREATE: false
-
     steps:
-    - uses: actions/checkout@v2
+    #----------------------------------------------
+    #       check-out repo and set-up python
+    #----------------------------------------------
+    - name: Check out repository
+      uses: actions/checkout@v2
     - name: Set up Python 3.7
       uses: actions/setup-python@v2
       with:
         python-version: 3.7
+    #----------------------------------------------
+    #  -----  install & configure poetry  -----
+    #----------------------------------------------
+    - name: Install Poetry
+      uses: snok/install-poetry@v1.2
+      with:
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+        installer-parallel: true
+
+    #----------------------------------------------
+    #       load cached venv if cache exists
+    #----------------------------------------------
     - name: Set up Poetry cache for Python dependencies
+      id: cached-poetry-dependencies
       uses: actions/cache@v2
-      if: startsWith(runner.os, 'Linux')
       with:
-        path: ~/.cache/pypoetry
-        key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
-        restore-keys: ${{ runner.os }}-poetry-
-    - name: Install Poetry
-      run: |
-        curl -fsS -o get-poetry.py https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py
-        python get-poetry.py -y
-        echo "$HOME/.poetry/bin" >> $GITHUB_PATH
-        rm get-poetry.py
+        path: .venv
+        key: venv2-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+    #----------------------------------------------
+    # install dependencies if cache does not exist
+    #----------------------------------------------
     - name: Install dependencies
-      run: poetry install --no-interaction
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root
     - name: Format with black
       run: |
+        source .venv/bin/activate
         python -m black --check .
```

i think that instead of the activate, you can do 
```suggestion
        poetry run python -m black --check .
```

---

I never use poetry envs, but I'm basing this off [the poetry GH workflow](https://github.com/python-poetry/poetry/blob/master/.github/workflows/main.yml#L86)

---

Let's try it!